android/src/main/java/com/tailscale/ipn/App.kt

```diff
@@ -464,4 +471,27 @@ class App : Application(), libtailscale.AppContext {
       }
     }
   }
+
+  fun prepareDownloadsFolder(): File {
+    var downloads = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS)
```

Comment:
nit for now: looks like `getExternalStoragePublicDirectory` was deprecated in API level 29. Some discussion [here](https://stackoverflow.com/questions/56468539/getexternalstoragepublicdirectory-deprecated-in-android-q) on replacements.